server/app/assets/javascripts/admin_predicate_configuration.ts

```diff
@@ -0,0 +1,356 @@
+          } else {
+            valueInput.setAttribute('step', '.01')
+            valueInput.setAttribute('placeholder', '$0.00')
+            valueInput.setAttribute('type', 'number')
+          }
+          break
+        case 'LONG':
+          if (
+            operatorValue.toUpperCase() === 'IN' ||
+            operatorValue.toUpperCase() === 'NOT_IN'
+          ) {
+            // IN and NOT_IN operate on lists of longs, which must be entered as a comma-separated list
+            valueInput.setAttribute('type', 'text')
+          } else {
+            valueInput.setAttribute('step', '1')
+            valueInput.setAttribute('placeholder', '0')
+            valueInput.setAttribute('type', 'number')
+          }
+          break
+        case 'DATE':
+          valueInput.setAttribute('type', 'date')
+          break
+        default:
+          valueInput.setAttribute('type', 'text')
+      }
+    }
+  }
+
+  /** Add a value row by duplicating the last row in the list and updating its group IDs. */
+  predicateAddValueRow(event: Event) {
+    event.preventDefault()
+    event.stopPropagation()
+
+    const currentRows = document.querySelectorAll('.predicate-config-value-row')
+    const templateRow = currentRows[currentRows.length - 1]
+    const newRow = templateRow.cloneNode(true) as HTMLElement
+
+    newRow.querySelectorAll('input').forEach((el: HTMLInputElement) => {
+      if (el.type === 'checkbox' || el.type === 'radio') {
+        el.checked = false
+      } else {
+        el.value = ''
+      }
+
+      const groupNumString = assertNotNull(el.name.match(/group-(\d+)/))[1]
+
+      let groupNum = parseInt(groupNumString, 10)
+      el.name = el.name.replace(/group-\d+/, `group-${++groupNum}`)
+      const newId = `${el.id}-${groupNum}`
+      el.id = newId
+
+      el.closest('label')?.setAttribute('for', newId)
+    })
+
+    const deleteButtonDiv = assertNotNull(
+      newRow.querySelector('.predicate-config-delete-value-row'),
+    ) as HTMLElement
+
+    deleteButtonDiv.addEventListener('click', (event: Event) =>
+      this.predicateDeleteValueRow(event),
+    )
+    assertNotNull(deleteButtonDiv.querySelector('svg')).classList.remove(
+      'hidden',
+    )
+
+    assertNotNull(
+      document.getElementById('predicate-config-value-row-container'),
+    ).append(newRow)
+  }
+
+  /**
+   * Delete the value row that includes the event target.
+   * */
+  predicateDeleteValueRow(event: Event) {
+    event.preventDefault()
+    event.stopPropagation()
+
+    const valueRow = (event.target as HTMLElement).closest(
+      '.predicate-config-value-row',
+    )
+
+    if (valueRow == null) {
+      throw new Error('Parent with class .predicate-config-value-row not found')
+    }
+
+    valueRow.remove()
+  }
+
+  /**
+   * Filter the operators available for each scalar type based on the current scalar selected.
+   *   @param {HTMLSelectElement} scalarDropdown The element to filter the operators for.
```

Comment:
Remove the period or add it to the others, I'm not sure what the style is for TS

Reply:
looks like the tendency is to have them: https://google.github.io/styleguide/jsguide.html

added here and on the similar lines in this file